src/test/util/setup_common.cpp

```diff
@@ -98,7 +98,9 @@ std::ostream& operator<<(std::ostream& os, const uint256& num)
 }
 
 BasicTestingSetup::BasicTestingSetup(const std::string& chainName, const std::vector<const char*>& extra_args)
-    : m_path_root{fs::temp_directory_path() / "test_common_" PACKAGE_NAME / g_insecure_rand_ctx_temp_path.rand256().ToString()},
+    : m_path_root{std::getenv("TEST_BITCOIN_PATH") ?
+                      std::getenv("TEST_BITCOIN_PATH") :
```

Comment:
pretty sure this will break when running in parallel.

What about a --nocleanup option, like the one in the functional test framework, if the goal is to not delete the dir?

Reply:
Not sure if this would work, but what about checking the datadir lock before removing it, e.g. with `util/system.cpp::LockDirectory`?

Reply:
To explain, tests running in parallel can not share a data directory. With rand256, each test gets a unique one. However, with `TEST_BITCOIN_PATH` all tests get the same one.

Reply:
I know. My suggestion was rather: since this is an advanced setting, we could let the user handle making sure that there are no tests running in parallel, but just add a simple check in case they aren't aware/mistaken (by checking the datadir lock)?

Reply:
> What about a --nocleanup option ... if the goal is to not delete the dir?

Yes, but also to have a fixed data directory path, so I can open its `debug.log` in an editor, and re-read the file as the test runs and after the test completes. (Or better yet, set the editor to automatically re-read if possible, which I do with `vim`.) Then if I run the test again, I can re-read the file again, without having to open a different file. 

But I think specifying this environment variable can also indicate not to delete the directory when the test completes; you wouldn't want to do this normally because the disk usage would continuously increase as you run the test repeatedly. If instead you specify a fixed directory, then disk usage doesn't grow as you repeatedly run tests, so it's okay for the test not to delete it at the end.

> tests running in parallel can not share a data directory

Yes, you definitely wouldn't specify this to multiple tests running in parallel. I would only use this when I'm running a single test that I'm trying to understand or debug. That is, I definitely would not set this environment variable in my `.bashrc`! (I could improve the doc changes to make that clear.) The default is to use separate directories (as today). 

> ... checking the datadir lock ...

The unit tests don't create a lock file; only `bitcoind` does that. The unit tests don't actually create a full datadir, only some parts of it, and can also create extra files that aren't normally in a datadir. (That's why I called it a working dir in the title and description, instead of a datadir, although the test `README` does call it a datadir.)

I don't think it's worth adding a lock file; I think users of this will be advanced enough to understand that multiple tests can't simultaneously share the same work directory.

Reply:
ok, sgtm